front-end/src/views/Meal.jsx

```diff
@@ -0,0 +1,87 @@
+import './Meal.css'
+import { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { baseUrl } from '../apiurl'
+import { BeatLoader } from 'react-spinners'
+
+export function Meal() {
+  const { mealId } = useParams()
+  const [mealData, setMealData] = useState(undefined)
+
+  const fetchMealDetails = async () => {
+    if (mealData) return
```

Comment:
you can also check whether you have the `mealId`. This could happen if someone would try to modify the id in the URL directly